### DISCUSSION TITLE
Are people still using Garmin devices with good results?

### DISCUSSION THREAD

**Discussion by lundstrj** [OpenSeizureDetector/Android_Pebble_SD#157] [04_Devices]
I'm asking because apart from initial success with a Venu SQ which allowed me to finally sleep after years of, well, not. It has been problem after problem related to Garmin pushing new firmware or Garmin Connect updates which has broken the OSD integration.

Mr Graham has been wonderful in debugging and helping out but at the end of the day, he cannot fix Garmin's code.

Out of desperation I got a Venu SQ 2 instead which worked great for one or two nights, until Garmin pushed an over the air update which put it on par with the flaky Venu SQ we had from before.

I'm now alternating a BangleJS and PineTime, which so far have been superior in stability and connectivity, albeit perhaps with problematic HRM on the BangleJS.

Have I've just had rotten luck with my Garmin devices (and then equally rotten luck with regards to the responsiveness from Garmin related to these issues)?

**Comment by pmithrandir** [comment]
Hi,

I would like to know more about what's behind "which put it on par with the flaky Venu SQ we had from before."

What difficulties are your encountering ?

I saw issues with battery, some small bugs related to the code of the osd garmin software... but nothing that could prevent the system to work at all.

Would you mind sharing more details please ?

Best regards,
Pierre

**Reply by jones139** [reply]
@pmithrandir, @lundstrj  is referring to this issue:  https://github.com/OpenSeizureDetector/Garmin_SD/issues/26
After a while the watch stops sending data to the phone, until you re-boot the watch.  I introduced the 'low data mode' to try to improve this, but we have not found the reason for it - It got much worse after a Garmin firmware update.

The only thing I have not tried which might help is to move away from using http to send data from the watch to the phone and change to using the Garmin messaging API to send data to the phone instead.

**Reply by pmithrandir** [reply]
thank you for the context.

Maybe all the changes that have been made or are ongoing could improve the situation. Some components were heavily modified and some exit condition were enhanced.

**Reply by lundstrj** [reply]
Sorry for the lack of context (re-reading what I wrote, it is a far cry from even beginning to make sense without the context). I suppose I assumed it was a more widespread problem as I had it on two separate Garmin models in exactly the same manner. In my mind, I figured it was more a matter of time before Garmin broke the OSD integration on more devices.

But yes, Graham is correct, that is exactly what I am referring to.

On an unrelated topic, I just realized I should have opened an issue, not a discussion about a problem I encountered with the PineTimeSD.

(It has been a rough couple of nights. I'll shape up once I get some shut-eye)

**Comment by jones139** [comment]
We are using an old VivoactiveHR for our 'production' set-up.    We only use it on a night, and find it does stop communicating with the phone during the day and the watch might need re-starting on an evening when we set it going.  I have always thought it is something to do with the charger location being out of bluetooth range of Benjamin's phone.   The problem occurs sufficiently infrequently that we just live with it.
